apps/angular/src/core/guards/auth.guard.ts

```diff
@@ -0,0 +1,32 @@
+import { inject } from '@angular/core';
+import { CanActivateFn, Router } from '@angular/router';
+
+import { AuthService } from '../services/auth.service';
+
+/** Guard provides user should be authenticated . */
+export function shouldBeAuthenticatedGuard(): CanActivateFn {
+	const authService = inject(AuthService);
+	const router = inject(Router);
+
+	const isAuthenticated = authService.isAuthenticated();
+
+	if (!isAuthenticated) {
+		router.navigate(['auth']);
```

Comment:
We can improve this code a bit. Look closely to return type of 'CanActivateFn'.
It should return either `boolean` or `UrlTree`.
`UrlTree` is used in guards to perform redirects, that's exactly what we need here.
To convert 'string' to `UrlTree` you can use 'parseUrl' utility from Router.

```suggestion
		return () => router.parseUrl('auth');
```